library/Icinga/Application/Config.php

```diff
@@ -350,11 +350,18 @@ public function saveIni($filePath = null, $fileMode = 0660)
             throw new LogicException('You need to pass $filePath or set a path using Config::setConfigFile()');
         }
 
-        if (! file_exists($filePath)) {
-            File::create($filePath, $fileMode);
-        }
+        $writer = $this->getIniWriter($filePath, $fileMode);
+        if ($writer->render() !== '') {
```

Comment:
Since IniWriter::write() does also call IniWriter::render() we should probably cache what's been rendered to avoid doing it twice.